src/Microsoft.Sbom.Api/Exceptions/ComponentDetectorException.cs

```diff
@@ -26,6 +26,9 @@ public ComponentDetectorException(string message, Exception innerException)
     {
     }
 
+#if NET8_0_OR_GREATER
+    [Obsolete(DiagnosticId = "SYSLIB0051")] // SerializationInfo type is obsolete as of .NET8
```

Comment:
Documentation on this error suggests this as the fix if you NEED the exception to be serializable. I after testing I couldn't find any issues by removing the [Serializable] attribute and the corresponding method. Could use suggestions on which approach to take in case I'm missing something.

Reply:
Reading the documentation[^1], it seems like we can likely drop this altogether:

> If you created a custom type derived from [System.Exception](https://learn.microsoft.com/en-us/dotnet/api/system.exception), consider whether you really need it to be serializable. It's likely that you don't need it to be serializable, as exception serialization is primarily intended to support remoting, and support for remoting was dropped in .NET Core 1.0. 

[^1]: https://learn.microsoft.com/en-us/dotnet/fundamentals/syslib-diagnostics/syslib0051